src/LiquidityPool.sol

```diff
@@ -462,10 +464,38 @@ contract LiquidityPool is Initializable, OwnableUpgradeable, UUPSUpgradeable, IL
         emit Rebase(getTotalPooledEther(), eETH.totalShares());
     }
 
+    function payProtocolFees(uint128 _protocolFees) public {
+        if (msg.sender != address(etherFiAdminContract) && msg.sender != address(this)) revert IncorrectCaller();
```

Comment:
this `msg.sender != address(this)` is a littly janky

I would make `_payProtocolFees` an internal function
And have an external version 
```
payProtocolFees() external { 
  _payProtocolFees()
}
```
And call the internal version from the other function that calls this method. And this would match how he handle this situation in our other contracts

Reply:
agreed. @vvalecha519 https://docs.soliditylang.org/en/latest/control-structures.html#internal-function-calls